pkg/operator/sync.go

```diff
@@ -665,7 +665,7 @@ func newContainers(config *OperatorConfig, features map[string]bool) []corev1.Co
 	// --feature-gates=<name>=<bool>,<name>=<bool>... arg
 	featureGateArgs := append(args, buildFeatureGatesString(features))
 
-	machineControllerArgs := append([]string{}, args...)
+	machineControllerArgs := append([]string{}, featureGateArgs...)
```

Comment:
This will break any provider that doesn't currently have the gate flag enabled as a no-op

Which platforms still need that gating no-op?

/hold until all platforms have the gating

Reply:
I think there are a few. I'll work on those today. 

I was thinking this change (passing the args from the operator) could be a separate PR. 

This way we can unblock the machine controller changes.